### PR TITLE
Add empty dashboard integration test

### PR DESCRIPTION
**Pull request by unclejustin**
Closes https://github.com/sourcegraph/sourcegraph/issues/41800

- Use puppeteer
- Mock data for dashboard with no insights
- Go to `/insights/dashboards/TEST_ID`

## Assertions
- [x] Dashboard select renders with test dashboard name
- [x] Add insight card renders

## Test plan

Integration tests should pass

Run locally

In one terminal `sg start web-standalone`
In another `sg test web-integration:debug lient/web/src/integration/insights/insight-dashboard.test.ts`

## App preview:

- [Web](https://sg-web-insights-integration-test-empty.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-usumldnrou.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboard-select/components/menu-button/MenuButton.tsx [hunk 1]
```diff
@@ -21,10 +21,14 @@ interface MenuButtonProps {
...
 export const MenuButton: React.FunctionComponent<React.PropsWithChildren<MenuButtonProps>> = props => {
-    const { dashboards, className } = props
+    const { dashboards, className, ...attributes } = props
 
     return (
-        <ListboxButton id="insight-dashboard-select-button" className={classNames(styles.button, className)}>
+        <ListboxButton
+            id="insight-dashboard-select-button"
+            className={classNames(styles.button, className)}
+            {...attributes}
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboard-select/components/menu-button/MenuButton.tsx:30]: Spreading attributes to get access to `data-testid`

File: client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboard-select/DashboardSelect.tsx [hunk 2]
```diff
@@ -82,7 +82,7 @@ export const DashboardSelect: React.FunctionComponent<React.PropsWithChildren<Da
                 value={value ?? 'unknown'}
                 onChange={handleChange}
             >
-                <MenuButton dashboards={rawDashboards} />
+                <MenuButton dashboards={rawDashboards} data-testid="dashboard-select-button" />
```
Comment (vovakulikov) [inline client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboard-select/DashboardSelect.tsx:85]: @unclejustin what about having here some semantic selector instead of data-test id?

File: client/web/src/integration/insights/dashboards/render-empty-dashboard.test.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+        await driver.page.goto(driver.sourcegraphBaseUrl + '/insights/dashboards/EMPTY_DASHBOARD')
+
+        const button = await driver.page.waitForSelector('[data-testid="dashboard-select-button"')
+
+        assert(button)
+
+        const buttonText: string = (await driver.page.evaluate(button => button.textContent, button)) as string
+
+        assert(/empty dashboard/i.test(buttonText))
```
Comment (vovakulikov) [inline client/web/src/integration/insights/dashboards/render-empty-dashboard.test.ts:50]: @unclejustin what about adding another case here for the empty dashboard scenario? Like empty dashboard should have create insight button, and this button should open "Add insight to " modal?

Comment (unclejustin) [reply client/web/src/integration/insights/dashboards/render-empty-dashboard.test.ts:50]: That's actually in the test cases that I forgot. Pulled the trigger too early for review on this one apologies.

File: client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboard-select/DashboardSelect.tsx (hunk 2 above)
Comment (unclejustin) [reply client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboard-select/DashboardSelect.tsx:85]: @vovakulikov I spent more time than I care to admit trying to do that very thing.

TLDR: Puppeteer selectors make this inordinately difficult. It's why I brought up upgrading puppeteer and using dom-testing-library's puppeteer extension.

File: client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboards-content/components/empty-insight-dashboard/EmptyInsightDashboard.tsx [hunk 4]
```diff
@@ -71,6 +71,7 @@ export const EmptySettingsBasedDashboard: React.FunctionComponent<
                 onClick={onAddInsight}
                 variant="secondary"
                 className="p-0 w-100 border-0"
+                data-testid="add-insights-button-card"
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/pages/dashboards/dashboard-page/components/dashboards-content/components/empty-insight-dashboard/EmptyInsightDashboard.tsx:74]: More compensating for puppeteer's poor selectors.

File: client/web/src/integration/insights/dashboards/render-empty-dashboard.test.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        assert(dashboardSelectButton)
+
+        const dashboardSelectButtonText: string = (await driver.page.evaluate(
+            button => button.textContent,
+            dashboardSelectButton
+        )) as string
+        const addInsightsButtonCardText: string = (await driver.page.evaluate(
+            button => button.textContent,
+            addInsightsButtonCard
+        )) as string
```
Comment (unclejustin) [inline client/web/src/integration/insights/dashboards/render-empty-dashboard.test.ts:56]: All of this code makes me sad 😢 but until we have better selectors, I think this is better than the `waitForFunction` alternatives.

**Comment by vovakulikov** [comment]
@unclejustin have you forgot to push local changes to this PR? description of this PR says "Add insight card renders" but I can't find this in this PR changes

**Comment by vovakulikov** [comment]
Oh I see, I messed different insight card. 🤦  I see that you added text assertion but do you think we also should test here logic as well? like that something really happens then you click this empty insight card button?

**Comment by unclejustin** [comment]
> Oh I see, I messed different insight card. 🤦 I see that you added text assertion but do you think we also should test here logic as well? like that something really happens then you click this empty insight card button?

Not yet. This is just the "does it render" smoke test. We will add to this over time.

The next test is that it renders with actual insights.

Then we start breaking down the interactivity.
